app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt

```diff
@@ -309,14 +309,19 @@ private fun assertSearchEngineURL(searchEngineName: String) {
         .check(matches(withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
 }
 
-private fun assertSearchEngineResults(minCount: Int) {
-    mDevice.waitForObjects(
-        searchSuggestionsList.getChild(UiSelector().index(minCount))
-    )
-
-    assertTrue(searchSuggestionsList.childCount >= minCount)
+private fun assertSearchEngineResults(rule: ComposeTestRule, searchSuggestion: String, searchEngineName: String) {
+    rule.waitUntil(waitingTime, waitForSearchSuggestions(rule, searchSuggestion, searchEngineName))
+    rule.onNodeWithText(searchSuggestion).assertIsDisplayed()
 }
 
+private fun waitForSearchSuggestions(rule: ComposeTestRule, searchSuggestion: String, searchEngineName: String): () -> Boolean =
+    {
+        rule.waitForIdle()
+        mDevice.waitForObjects(mDevice.findObject(UiSelector().textContains(searchSuggestion)))
+        rule.onAllNodesWithTag("mozac.awesomebar.suggestion").assertAny(hasText(searchSuggestion) and hasText(searchEngineName))
+        mDevice.findObject(UiSelector().textContains(searchSuggestion)).waitForExists(waitingTime)
```

Comment:
I'm not sure if I fully understand this method, but just in case, double checking if lines 320 and 322 aren't doing the same thing.

Reply:
To be able to use `rule.waitUntil(timeoutMillis, condition: () -> Boolean)` I had to create a lambda function that returns a boolean, in this case `waitForSearchSuggestions`

On 319 I wait for the compose to be idle.
On 320 I use `waitForObjects()` to **wait** and assert that the search suggestions are loaded and displayed.

A node looks like this:
```
Node found:
Node #33 at (l=0.0, t=66.0, r=1080.0, b=220.0)px, Tag: 'mozac.awesomebar.suggestion'
Focused = 'false'
Text = '[mozilla, DuckDuckGo]'
Actions = [OnClick, GetTextLayoutResult]
MergeDescendants = 'true'

```

On 321 I want to assert that the node with tag [mozac.awesomebar.suggestion](https://github.com/mozilla-mobile/android-components/blob/main/components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/internal/Suggestion.kt#L55) contains both the search suggestion and the previously selected search engine.

On 322 I need to return a boolean to be able to use `waitForSearchSuggestions` in `rule.waitUntil()`

Reply:
You both know more than me about Compose testing so I trust the decision here.